File: src/main/java/org/trp/shincolle/entity/AircraftAiNumbers.java
package org.trp.shincolle.entity;

final class AircraftAiNumbers {
    static final int LIFETIME_TICKS = 1200;
    static final int HOST_CHECK_TIMEOUT = 20;
    static final int INITIAL_BOOST_DURATION = 34;
    static final double INITIAL_BOOST_SPEED = 0.375D;
    static final double INITIAL_BOOST_Y = 0.1D;
    static final int TARGETING_INTERVAL = 16;
    static final int RETURN_HOME_CHECK_INTERVAL = 16;
    static final int RETURN_HOME_STUCK_TICK_LIMIT = 120;
    static final int RETURN_HOME_FAILSAFE_TICKS = 20 * 30;
    static final int RETURN_HOME_TELEPORT_COOLDOWN_TICKS = 100;
    static final double RETURN_HOME_TELEPORT_DISTANCE_SQ = 256.0D;
    static final double TARGETING_RANGE_NORMAL = 24.0D;
    static final double TARGETING_RANGE_AIR_ONLY = 32.0D;

    static final float ATTACK_RANGE_LIGHT = 6.0F;
    static final float ATTACK_RANGE_HEAVY = 16.0F;
    static final double RAND_POS_MIN_LIGHT = 4.5D;
    static final double RAND_POS_RAND_LIGHT = 1.5D;
    static final double RAND_POS_MIN_HEAVY = 12.0D;
    static final double RAND_POS_RAND_HEAVY = 4.0D;

    static final double DEATH_GRAVITY = 0.08D;
    static final int DEATH_TIME_BURNING = 30;
    static final int DEATH_TIME_EXPLOSION = 90;
    static final int AMMO_RETURN_PENALTY_LIGHT = 3;
    static final int AMMO_RETURN_PENALTY_HEAVY = 1;
    static final int INITIAL_AMMO_LIGHT = 9;
    static final int INITIAL_AMMO_HEAVY = 3;
    static final int BASE_ATTACK_SPEED_AIRCRAFT = 100;
    static final int FIXED_ATTACK_DELAY_AIRCRAFT = 35;

    static final int ATTACK_ACTIVATION_TICKS = 20;
    static final int ATTACK_BEHAVIOR_PRIORITY = 1;
    static final int ATTACK_RECALC_INTERVAL_MASK = 0xF;
    static final double ATTACK_TARGET_Y_OFFSET = 2.0D;
    static final double ATTACK_SPEED_SLOW = 0.3D;
    static final double ATTACK_SPEED_FAST = 0.6D;
    static final double RETURN_HOME_SPEED = 0.5D;
    static final double RETURN_HOME_EXTRA_HEIGHT = 1.0D;
    static final double RETURN_HOME_ARRIVAL_BASE = 2.0D;
    static final double RETURN_HOME_TELEPORT_EXTRA = 0.75D;
    static final int RANDOM_CRUISE_ATTEMPTS = 25;
    static final float RANDOM_CRUISE_ANGLE_STEP = 15.0F;
    static final double RANDOM_CRUISE_Y_OFFSET = 2.0D;
    static final double RANDOM_CRUISE_Y_RANDOM = 2.0D;
    static final double RANDOM_CRUISE_FALLBACK_Y = 5.0D;

    private AircraftAiNumbers() {
    }
}

File: src/main/java/org/trp/shincolle/entity/AircraftAiNumbers.kt
package org.trp.shincolle.entity

internal object AircraftAiNumbers {
    const val LIFETIME_TICKS: Int = 1200
    const val HOST_CHECK_TIMEOUT: Int = 20
    const val INITIAL_BOOST_DURATION: Int = 34
    const val INITIAL_BOOST_SPEED: Double = 0.375
    const val INITIAL_BOOST_Y: Double = 0.1
    const val TARGETING_INTERVAL: Int = 16
    const val RETURN_HOME_CHECK_INTERVAL: Int = 16
    const val RETURN_HOME_STUCK_TICK_LIMIT: Int = 120
    val RETURN_HOME_FAILSAFE_TICKS: Int = 20 * 30
    const val RETURN_HOME_TELEPORT_COOLDOWN_TICKS: Int = 100
    const val RETURN_HOME_TELEPORT_DISTANCE_SQ: Double = 256.0
    const val TARGETING_RANGE_NORMAL: Double = 24.0
    const val TARGETING_RANGE_AIR_ONLY: Double = 32.0

    const val ATTACK_RANGE_LIGHT: Float = 6.0f
    const val ATTACK_RANGE_HEAVY: Float = 16.0f
    const val RAND_POS_MIN_LIGHT: Double = 4.5
    const val RAND_POS_RAND_LIGHT: Double = 1.5
    const val RAND_POS_MIN_HEAVY: Double = 12.0
    const val RAND_POS_RAND_HEAVY: Double = 4.0

    const val DEATH_GRAVITY: Double = 0.08
    const val DEATH_TIME_BURNING: Int = 30
    const val DEATH_TIME_EXPLOSION: Int = 90
    const val AMMO_RETURN_PENALTY_LIGHT: Int = 3
    const val AMMO_RETURN_PENALTY_HEAVY: Int = 1
    const val INITIAL_AMMO_LIGHT: Int = 9
    const val INITIAL_AMMO_HEAVY: Int = 3
    const val BASE_ATTACK_SPEED_AIRCRAFT: Int = 100
    const val FIXED_ATTACK_DELAY_AIRCRAFT: Int = 35

    const val ATTACK_ACTIVATION_TICKS: Int = 20
    const val ATTACK_BEHAVIOR_PRIORITY: Int = 1
    const val ATTACK_RECALC_INTERVAL_MASK: Int = 0xF
    const val ATTACK_TARGET_Y_OFFSET: Double = 2.0
    const val ATTACK_SPEED_SLOW: Double = 0.3
    const val ATTACK_SPEED_FAST: Double = 0.6
    const val RETURN_HOME_SPEED: Double = 0.5
    const val RETURN_HOME_EXTRA_HEIGHT: Double = 1.0
    const val RETURN_HOME_ARRIVAL_BASE: Double = 2.0
    const val RETURN_HOME_TELEPORT_EXTRA: Double = 0.75
    const val RANDOM_CRUISE_ATTEMPTS: Int = 25
    const val RANDOM_CRUISE_ANGLE_STEP: Float = 15.0f
    const val RANDOM_CRUISE_Y_OFFSET: Double = 2.0
    const val RANDOM_CRUISE_Y_RANDOM: Double = 2.0
    const val RANDOM_CRUISE_FALLBACK_Y: Double = 5.0
}
